### ISSUE TITLE
Allow me to disable automatic location requests

### ISSUE THREAD

**Issue by jplatte** [organicmaps/organicmaps#3138]
When I start Organic Maps, currently it tries to get my current location. In case location info is disabled in the system settings, an annoying popup opens, but even when it is enabled and I'm not annoyed by it, I am not happy about this behavior.

In many cases when opening up the app the current location is irrelevant, for example: I want to look up the address of a location where I could stay on a trip, to see points of interest nearby.

I would much prefer if Organic Maps only tried to obtain my current location upon explicit request, or continuously while in navigation mode. I would argue that it doesn't really impact convenience if the current behavior is just changed altogether, but if changing that leads to a worse experience for others, maybe a compromise could be a setting?

Thanks for maintaining this great application!

Related: #1128

**Comment by biodranik** [comment]
1. Android or iOS?
2. The previous mode should be stored and restored on startup. There should be no annoying popup. 
3. Your current location is critically important for the search algorithm to work properly.
4. Changing the behavior breaks many user scenarios. We got a lot of negative feedback after recent changes on Android.

**Comment by jplatte** [comment]
> 1. Android or iOS?

I'm talking about Android. 

> 2. The previous mode should be stored and restored on startup. There should be no annoying popup.

I'm not sure what you mean by 'mode', but I'm still seeing the popup on every launch. I'm pretty sure this was discussed somewhere before, but it seems like it wasn't on #1128.
 
> 3. Your current location is critically important for the search algorithm to work properly.

Is this documented somewhere? I would like to learn more about it.

**Comment by biodranik** [comment]
Can you please record a video of that popup on every launch? We definitely fixed it some months ago.

Search should prioritize results using some criteria. Distance from you to the search result is one of the most important criteria.

**Comment by jplatte** [comment]
https://user-images.githubusercontent.com/951129/194064226-f8fac563-65d5-4f9e-9c8b-fa9fc84e03cf.mp4

**Comment by biodranik** [comment]
1. What happens if you wait until the location is not found, and then close the app?
2. What happens if you disable location services, launch OM, and close the app? It won't start a location search, right?
3. Also check related #3450

**Comment by endim8** [comment]
you need to update OrganicMaps, that looks like an old version

**Comment by jplatte** [comment]
The screencast is old. I couldn't originally upload it until I cut it today. I'm now on Version 2022.09.22-3-FDroid.

2\. is what I did in that screencast. It does a location search, at least it animates the location button. If I wait for the animation to stop and then re-launch the app, the behavior is still the same.

**Comment by jplatte** [comment]
Maybe it's worth noting that I'm on LineageOS w/o Google Apps and so the only location that's available on a system level should be GPS, AFAIK.

**Comment by biodranik** [comment]
Strangely, if location services are disabled in the system, I see the dialog once, but it doesn't appear again for me after the app restart. This is the correct behavior that we implemented about 4-6 months ago. How is it possible, that it doesn't behave the same on your device?

**Comment by Gnorv** [comment]
I only get this notification sometimes, but even so it is often enough to annoy me. I am not sure when it appears and when not.
How about only displaying the request to enable if two conditions are met at the same time:
1. System wide location services are off
2. User clicks on location button
If the services are on, Organic Maps can just search location no matter what user does. If they are off but the user does not ask for his location, why should Organic Maps assume the user wants to see his location?

**Comment by jplatte** [comment]
I just noticed I'm no longer getting the error consistently. Maybe the logic of only showing it when location was available at the last run but isn't now is properly working now.

I would personally still prefer if it only showed when I request the location explicitly, but I can understand the rationale behind what you described should be happening.

**Comment by biodranik** [comment]
We have added an additional icon for the location button, so now it should be clear when location services are not enabled, and the location is not searched. Please test how the next release works and let us know if you find any issues.

**Comment by jplatte** [comment]
What additional icon do you mean? I see a strike through the icon now, is that what you mean?

**Comment by biodranik** [comment]
Yes.

**Comment by jplatte** [comment]
I think that's nice, but I don't see how it's directly related to this issue.

To re-iterate my current understanding: When the app is started, it tries to get the user's location if the location functionality was used on the last run, regardless of whether the system-wide location switch is on of off. If this fails (e.g. because the system-wide location switch is off), an error is displayed. Personally I would want the app to trust me that I don't want location if I have the system-wide setting turned off, and only display an error when I actually click the button in the UI, regardless of what happened on the last run of the app.

**Comment by biodranik** [comment]
The issue is that many users are get used to automatic location and they often complain to us that location doesn't work, without realizing that it is/was disabled. 

Now dialog should appear only once per 24 hours I believe, and only if the app was closed with active location.

**Comment by jplatte** [comment]
> Now dialog should appear only once per 24 hours I believe, and only if the app was closed with active location.

Which version should this contain this change? I'm using `2022.11.02-2-FDroid` and am still seeing the popup when opening after closing w/o location. However, the animation on the button doesn't happen in this case.

**Comment by biodranik** [comment]
@jplatte Strange. Can you please clarify and confirm your case?

1. Close OM
2. Disable location in system settings.
3. Start OM.
4. You see dialog "Enable disabled location in system settings" (you should not see it again if it was shown < 24h ago), and location icon is crossed, right? 
5. Close OM.
6. Open OM again.
7. You again see the same dialog (you should NOT see it)., and location icon is crossed. Right?

**Comment by jplatte** [comment]
Yes, just tried it again and that is exactly what happens. At step 4, the location button has the animated icon behind the popup, when I close and re-open (step 7), the popup appears with the button having the crossed non-animated icon.

**Comment by biodranik** [comment]
@rtsisyk can you please re-check this case?

**Comment by nach0** [comment]
Hi, I came here to add a request for this same popup. I am in the same situation as @jplatte and get the pop up asking to enable location services every single time I launch the app, regardless of previous state and how long ago I last used the app.

My idea was to suggest adding an option to "not ask to enable location services". That could be disabled by default (so it still asks, as I understand most people want), but those of us that use the app without location 90% of the time can select that option.

I am running the latest version 2022.11.24-3-FDroid on iodéOS 3.3 (based on Android 12).

**Comment by jplatte** [comment]
Yeah, seems to work. Thanks!

**Comment by nach0** [comment]
I am sorry, but in my case I keep getting the pop-up asking "Please enable Location Services" some times. It is not every time I launch the app, but sometimes it happens 2 consecutive times, sometimes it doesn't happen in 1 day or so. I haven't recognized a pattern here.
I am running the latest F-Droid version (2022.12.18-12-FDroid) on iodéOS 3.4 (based on Android 12).

**Comment by biodranik** [comment]
Please see #4390
